### ISSUE TITLE
Missing sh:ValidationReport when creating resources

### ISSUE THREAD

**Issue by PaulPawletta** [BlueBrain/nexus-python-sdk#91]
When I'm creating resources that are constrained by a schema I'm getting an exception that doesn't tell me what the validation issues are:

<img width="1033" alt="Screenshot 2020-09-11 at 17 49 19" src="https://user-images.githubusercontent.com/16834721/92946692-45e21d00-f457-11ea-97ef-5249e1f82ff8.png">

This propagates further to Nexus Forge when I'm creating a constrained resource:

<img width="1013" alt="Screenshot 2020-09-11 at 17 53 19" src="https://user-images.githubusercontent.com/16834721/92947005-b7ba6680-f457-11ea-8d11-a1cedd57eae9.png">

So basically, I never really know why the validation failed. 

Is there a reason why the elaborate sh:ValidationReport is omitted for nxs.resources.create()?

**Comment by MFSY** [comment]
Hi,
Thanks for using our tools and for opening this issue. The issue is referring to:
- nexus-python-sdk which is a wrapper around Nexus Delta REST API with some helpers 
- Nexus Forge which a python framework for ingesting , transforming, validating and accessing data that uses Nexus Delta as a store. 

It is true that Nexus Forge is still relying on nexus-python-sdk for some calls (e.g forge.register) to the the Nexus Delta API.

We recommend to the nexus-python-sdk's users to switch to Nexus Forge (for issues and feature request) as the feature set is richer there to reach better productivity with Nexus Delta for example.

Now about leveraging sh:ValidationReport for validation result. This is available in Nexus Forge through:

`forge.validate(resource)`

Note that:
* `forge.validate(resource)` is not based on nexus-python-sdk
* It performs an in-memory validation using [pyshacl](https://github.com/RDFLib/pySHACL). The resource is not persisted and a validation report is generated

Please have a look at [Nexus Forge's Modelling Jupyter notebook](https://github.com/BlueBrain/nexus-forge/blob/master/examples/notebooks/getting-started/11%20-%20Modeling.ipynb) demonstrating the usage of `forge.validate(resource)`. All Nexus forge Notebooks can be simply ran in [Binder](https://mybinder.org/v2/gh/BlueBrain/nexus-forge/master?filepath=examples%2Fnotebooks%2Fgetting-started) as explained in Nexus Forge's [getting started documentation](https://nexus-forge.readthedocs.io/en/latest/#getting-started).

Example of validation report generated by `forge.validate(resource)`

```
<action> _validate_one
<succeeded> False
<error> ValidationError: 
Validation Report
Conforms: False
Results (1):
Constraint Violation in NodeConstraintComponent (http://www.w3.org/ns/shacl#NodeConstraintComponent):
	Severity: sh:Violation
	Source Shape: this25:DatasetShape
	Focus Node: [ <https://neuroshapes.org/contribution> [ rdf:type <http://schema.org/Person> ] ; rdf:type <http://schema.org/Dataset> ]
	Value Node: [ <https://neuroshapes.org/contribution> [ rdf:type <http://schema.org/Person> ] ; rdf:type <http://schema.org/Dataset> ]
```

I hope this helps.

**Comment by PaulPawletta** [comment]
Hi @MFSY , thanks for the clarification. I will stick to the Nexus Forge library for the future then.

Just for my understanding forge.validate uses a local schema for in-memory validation right? 
Because how I ended up using forge.register and nxs.resource.create was, because I wanted to have the validation done against the persisted schema on Nexus Delta.

**Comment by MFSY** [comment]
`forge.validate` will use whatever schemas are loaded in the Model of the forge session. Schemas can be loaded from a local dir or remotely from Nexus Delta store (named BlueBrainNexus).

You can configure the forge session to pull schemas from Nexus Delta. Here is a configuration example to adapt to your context:

```
config['Model'] = {
    "name": "RdfModel",
    "origin": "store",
    "source": "BlueBrainNexus",
    # bucket: <when 'origin' is 'store', a Store bucket>
    "context": {
        "iri": "https://bbp.neuroshapes.org",
        "bucket": "neurosciencegraph/datamodels",
    },
}
```


Have a look at the [Nexus Forge's Initialization Jupyter notebook](https://github.com/BlueBrain/nexus-forge/blob/master/examples/notebooks/getting-started/00%20-%20Initialization.ipynb) for the different configuration options. You can view the [full list of configuration options](https://nexus-forge.readthedocs.io/en/latest/interaction.html#forge) in Nexus Forge documentation.

**Comment by PaulPawletta** [comment]
Great, thanks that works! I will close the issue.
